SparkPost/2web2ui
# FAD-5342 Templates Preview & Send View

_Refer to [FAD-5342](https://jira.int.messagesystems.com/browse/FAD-5342)_

Before:

<img width="1387" alt="screen shot 2018-01-03 at 1 38 14 pm" src="https://user-images.githubusercontent.com/1335605/34534236-68e7e65e-f08b-11e7-97d2-bbd1e83c63f3.png">

After:

<img width="1384" alt="screen shot 2018-01-03 at 1 36 10 pm" src="https://user-images.githubusercontent.com/1335605/34534180-338ffcc6-f08b-11e7-8ffa-4fa22f355a98.png">

How to test?

1. Navigate to the template list page, `/templates`
1. Create a new template or open an existing
1. Make sure your draft has different html and text than published and it has at least one variable (e.g. `Hello, {{name}}`)
1. Navigate to the "Test Data" tab, enter some `substitution_data` (e.g. `{ "name": "Bob Boolean" }`, click "Safe Draft" or "Publish Template" 
1. Click "Preview & Send" link
1. Enter a valid email address in the "To" input field
1. Click "Send Email"

The email on the preview page should look identical to the email in your inbox.

Todo:
- [x] Feature documentation for this PR
- [x] Wire-up "Send Email" button to send a transmission
- [x] Write tests for new actions creators
- [x] Submit follow-up ticket to move "Test Data" tab from edit pages to preview page ([FAD-6012](https://jira.int.messagesystems.com/browse/FAD-6012))
- [x] Submit follow-up ticket to fix webui and allow sending a test email with published template ([only uses draft](https://github.com/SparkPost/webui/blob/36283c11c8aafa826b18a219ed21d4615af9e8a4/src/app/templates/templates-edit-controller.js#L736)) and fix webui to not overwrite draft template when sending published template test email ([FAD-6013](https://jira.int.messagesystems.com/browse/FAD-6013))
- [x] Submit follow-up ticket to fix 2web2ui to not persist template test data after a user logs out ([FAD-6017](https://jira.int.messagesystems.com/browse/FAD-6017))
